routes/auth.router.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const {generateJWT} = require('../jwt');
+const bcrypt = require('bcryptjs');
+const router = express.Router();
+const bcryptSalt = bcrypt.genSaltSync(8);
+
+const User = require('../models/user.model');
+const Cart = require('../models/cart.model');
+const Wishlist = require('../models/wishlist.model');
+
+router.post('/signup', async function(req, res) {
+  const { email, password, name } = req.body;
+  const existingUser = await checkIfUserExists(email);
+  if (!existingUser) {
+    const hashedPassword = bcrypt.hashSync(password, bcryptSalt);   
+    const user = await User.create({ name: name, email: email, password: hashedPassword });
+    const cart = await Cart.create({uid: user._id});
+    const wishlist = await Wishlist.create({uid: user._id});
+    const token = generateJWT(user._id.toString());
+
+    res.json({
+      success: true,
+      token: token
+    });
+  } else {
+    res.status(409).json({
+      success: false,
+      error: {
+        message: "User Already Exists"
+      }
+    });
+  }
+});
+
+router.post('/login', async function(req, res){
+  const {email, password} = req.body;
+  const existingUser = await checkIfUserExists(email);
+  if(existingUser){
+    const validPassword = bcrypt.compareSync(password, existingUser.password);
+    if(validPassword){
+      const token = generateJWT(existingUser._id.toString());
+      res.json({
+        success: true,
+        token: token
+      })
+    }else{
+      res.status(401).json({
+        success: false,
+        error: {
+          message: "Invalid password"
+        }
+      });
+    }    
+  }else{
+    res.status(401).json({
+       success: false,
+        error: {
+          message: "User not registered"
+        }
+    })
+  }  
+})
+
+async function checkIfUserExists(email) {
+  const user = await User.find({ email: email });
+  return user[0];
+}
```

Comment:
instead if find you can use findOne right?

Reply:
yes findOne makes the right fit, changes have been made